cmd/api/grpc_client.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	pb "github.com/RHEcosystemAppEng/cluster-iq/generated/agent"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+const (
+	// grpcTimeoutSeconds defines the timeout in seconds for gRPC operations.
+	grpcTimeoutSeconds = 10
+)
+
+// APIGRPCClient manages the gRPC client connection and operations for the API server.
+// It provides methods to interact with the Agent service via gRPC.
+type APIGRPCClient struct {
+	// Client is the gRPC client used to communicate with the Agent service.
+	Client pb.AgentServiceClient
+	// CTX is the context used for gRPC operations.
+	CTX context.Context
+	// Cancel is the function to cancel the gRPC context.
+	Cancel context.CancelFunc
+	// logger is used for logging gRPC operations and errors.
+	logger *zap.Logger
+}
+
+// NewAPIGRPCClient initializes and returns a new APIGRPCClient.
+// It establishes a connection to the Agent service and sets up the gRPC client.
+//
+// Parameters:
+// - agentURL: The URL of the Agent service to connect to.
+// - logger: Logger instance for logging.
+//
+// Returns:
+// - A pointer to the initialized APIGRPCClient.
+// - An error if the connection cannot be established.
+func NewAPIGRPCClient(agentURL string, logger *zap.Logger) (*APIGRPCClient, error) {
+	// Initializing gRPC Client
+	conn, err := grpc.Dial(agentURL, grpc.WithInsecure())
```

Comment:
Deprecated
https://pkg.go.dev/google.golang.org/grpc@v1.69.4#Dial
Can be replaced with
```go
conn, err := grpc.NewClient(agentURL, grpc.WithTransportCredentials(insecure.NewCredentials()))
```